fem/estimators.hpp

```diff
@@ -396,6 +398,136 @@ class LpErrorEstimator : public ErrorEstimator
    virtual ~LpErrorEstimator() {}
 };
 
+#ifdef MFEM_USE_MPI
+/** @brief The KellyErrorEstimator class provides a fast error indication
+    for smooth scalar parallel problems.
+
+    A very simple error *indicator* based on the following papers.
+
+    Kelly, D. W., et al. "A posteriori error analysis and adaptive processes in
+    the finite element method: Part I—Error analysis." International journal for
+    numerical methods in engineering 19.11 (1983): 1593-1619.
+
+    De SR Gago, J. P., et al. "A posteriori error analysis and adaptive
+    processes in the finite element method: Part II—Adaptive mesh refinement."
+    International journal for numerical methods in engineering 19.11 (1983):
+    1621-1656.
+
+    It can be roughly described by:
+        ||∇(u-uₕ)||ₑ ≦ √( C ∑ₖ (hₖ ∫ |J[∇uₕ]|²) dS )
+    where "e" denotes an element, ||⋅||ₑ the corresponding local norm and k the
+    corresponding faces. u is the analytic solution and uₕ the discretized solution.
+    hₖ is a factor dependend on the element geometry. J is the jump function, i.e.
+    the difference between the limits at each point for each side of the face.
+    A custom method to compute hₖ can be provided. It is also possible to estimate
+    the error only on a subspace by feeding this class an attribute array describing
+    the subspace. The error on boundary faces is ignored.
+
+    @note This algorithm is only for Poisson problems a proper error esimator.
+    The current implementation does not reflect this, because the factor "C" is not
+    included.
+*/
+class KellyErrorEstimator final : public ErrorEstimator
+{
+private:
+   int current_sequence = -1;
+
+   Vector error_estimates;
+
+   double total_error = 0.0;
+
+   Array<int> attributes;
+
+   /** @brief The method to compute hₖ.
+
+       Defaults to hₖ=det(J)^(-dim)/2p. This is a slight variation of the
+       description by Bangerth.
+   */
+   std::function<double(ParMesh*, const int)> compute_element_coefficient = [](
+                                                                               ParMesh* pmesh, const int e)
+   {
+      // Obtain jacobian of the transformation.
+      DenseMatrix J;
+      // pmesh->GetElementJacobian(e, J); //..protected....
```

Comment:
I forgot to include this one in my questions. The code below this comment is basically a copy and paste of the `GetElementJacobian` method. I had to do this because the `GetElementJacobian` is marked protected and not public. Is there a reason for this? I am really unsure how to resolve this properly, but my suggestion would be to move `GetElementJacobian` from protected to public, because it can be useful outside of the mesh class.